### PR TITLE
Create FAQ page

### PR DESCRIPTION
**Pull request by Taraskin777**
## Ticket
Link to ticket:

https://teamchallengeonlinestore.atlassian.net/jira/software/projects/TOS/boards/1?assignee=712020%3A4102cab5-ba41-4622-b0e9-fca65c2e84c6&selectedIssue=TOS-150

## Description
Create FAQ page according to Figma's design

- Change 1
- Change 2

## Screenshots

Screenshot_7
Screenshot_8
Screenshot_10
Screenshot_9

## Checklist
- [x] Did you test it?
- [x] Did you add a screenshot?
- [x] Did you fix the merge conflict?
- [x] Did you fill the PR template?
- [x] Did you review this PR on your own?

#### Last step if the ABOVE STEPS are COMPLETED!
- [x] Did you share the link in a group chat?

### CODE REVIEW

**Review by dev1B — CHANGES_REQUESTED** [review]
image
Looks like you forgot complete it for desktop

**Comment by Taraskin777** [comment]
> image Looks like you forgot complete it for desktop

fixed

File: app/components/BurgerMenu/BurgerMenu.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const BurgerMenu = () => {
           <nav className={styles.nav}>
             <Link href="/">Про компанію</Link>
             <Link href="/">Новини та акції</Link>
-            <Link href="/">FAQ</Link>
+            <Link href="/faq">FAQ</Link>
```
Comment (dev1B) [inline app/components/BurgerMenu/BurgerMenu.tsx:60]: 👍

File: app/components/Faq/Accordion.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import AccordionItem from './AccordionItem';
+
+interface FaqItem {
+  id: number;
+  question: string;
+  answer: string;
+}
+
+interface FaqListProps {
+  FaqList: FaqItem[];
```
Comment (dev1B) [inline app/components/Faq/Accordion.tsx:14]: camelcase, please

File: app/components/Faq/Accordion.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+const Accordion = ({ FaqList }: FaqListProps) => {
```
Comment (dev1B) [inline app/components/Faq/Accordion.tsx:17]: Could you move this component into the components folder and make it more generic ( types should not refer to FAQ)

File: app/components/Faq/AccordionItem.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  answer: string;
+}
+
+interface AccItem {
+  faqItem: FaqItem;
+  onClick: () => void;
+  isOpen: boolean;
+}
+
+const AccordionItem = ({ faqItem, onClick, isOpen }: AccItem) => {
```
Comment (dev1B) [inline app/components/Faq/AccordionItem.tsx:19]: Could you move this component into the components folder and make it more generic ( types should not refer to FAQ)

File: app/components/Faq/AccordionItem.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { useRef } from 'react';
+
+import { RiArrowDownSLine } from 'react-icons/ri';
+import cn from 'classnames';
+import styles from './Faq.module.scss';
+
+interface FaqItem {
```
Comment (dev1B) [inline app/components/Faq/AccordionItem.tsx:7]: This type is duplicated.

File: app/components/Faq/Faq.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    answer:
+      'Так, ми регулярно проводимо акції та знижки на окремі види рослин або під час святкових періодів. Слідкуйте за нашими оновленнями та спеціальними пропозиціями!',
+  },
+];
+
+const Faq = () => (
+  <section className={styles.section}>
+    <div className={styles.mainblock}>
+      <h2>FAQ</h2>
+      <p className={styles.title}>Тут ви знайдете відповіді на найпоширеніші запитання </p>
```
Comment (dev1B) [inline app/components/Faq/Faq.tsx:35]: subtitle mb?

File: app/components/Faq/Faq.module.scss [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  }
+}
+
+.collapse {
+  padding-right: 20px;
+  overflow: hidden;
+  transition: height 0.3s;
+}
+
+@media (width >= 768px) {
```
Comment (dev1B) [inline app/components/Faq/Faq.module.scss:84]: Could you create scss const for media/mixin

File: app/components/Faq/Faq.module.scss [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+      transition: all 0.3s;
+    }
+
+    &.rotate {
+      transform: rotate(0.5turn);
+      color: rgb(9 95 77 / 100%);
+    }
+  }
+
+  & p {
```
Comment (dev1B) [inline app/components/Faq/Faq.module.scss:61]: Could you use class?

**Review by dev1B — CHANGES_REQUESTED** [review]

File: app/components/Faq/Accordion.tsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  const [openId, setId] = useState<number | null>(null);
+
+  const handleAccordion = (id: number) => {
+    if (id === openId) {
+      setId(null);
+    } else setId(id);
+  };
+
+  return (
+    <ul>
```
Comment (dev1B) [inline app/components/Faq/Accordion.tsx:27]: ul?

File: app/components/Faq/Accordion.tsx (hunk 2 above)
Comment (Taraskin777) [reply app/components/Faq/Accordion.tsx]: +

Comment (Taraskin777) [reply app/components/Faq/Accordion.tsx]: +

File: app/components/Faq/Accordion.tsx (hunk 3 above)
Comment (Taraskin777) [reply app/components/Faq/Accordion.tsx]: +

File: app/components/Faq/AccordionItem.tsx (hunk 4 above)
Comment (Taraskin777) [reply app/components/Faq/AccordionItem.tsx]: fixed

File: app/components/Faq/AccordionItem.tsx (hunk 5 above)
Comment (Taraskin777) [reply app/components/Faq/AccordionItem.tsx]: fixed

File: app/components/Faq/Faq.tsx (hunk 6 above)
Comment (Taraskin777) [reply app/components/Faq/Faq.tsx]: fixed

File: app/components/Faq/Faq.module.scss [hunk 10]
```diff
@@ -0,0 +1,113 @@
+@import '@utils';
+
```
[the remark this answers is not in the document]
Comment (Taraskin777) [reply app/components/Faq/Faq.module.scss:2]: fixed

File: app/components/Faq/Faq.module.scss (hunk 7 above)
Comment (Taraskin777) [reply app/components/Faq/Faq.module.scss]: fixed

File: app/components/Faq/Faq.module.scss (hunk 8 above)
Comment (Taraskin777) [reply app/components/Faq/Faq.module.scss]: no

File: app/components/Faq/AccordionItem.tsx [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  });
+
+  const showAnswer =
+    isOpen && itemRef.current
+      ? { height: `${itemRef.current.scrollHeight + 40}px` }
+      : { height: '0px' };
+
+  return (
+    <li className={styles.oftenQa}>
+      <button className={questionClass} onClick={() => onClick()} type="button">
```
[the remark this answers is not in the document]
Comment (Taraskin777) [reply app/components/Faq/AccordionItem.tsx]: id doesn't work

File: app/components/Faq/Accordion.tsx (hunk 9 above)
Comment (Taraskin777) [reply app/components/Faq/Accordion.tsx]: ul

**Pull request merged**